packages/aws-cdk-lib/aws-synthetics/lib/canary.ts

```diff
@@ -513,11 +543,45 @@ export class Canary extends cdk.Resource implements ec2.IConnectable {
   }
 
   private createRunConfig(props: CanaryProps): CfnCanary.RunConfigProperty | undefined {
-    if (!props.environmentVariables) {
+    if (props.activeTracing === undefined &&
+      !props.environmentVariables &&
+      !props.memory &&
+      !props.timeout) {
       return undefined;
     }
+    const activeTracingNotSupportedRuntime = [
+      Runtime.SYNTHETICS_PYTHON_SELENIUM_2_1,
+      Runtime.SYNTHETICS_PYTHON_SELENIUM_3_0,
+    ];
+
+    if (props.activeTracing && activeTracingNotSupportedRuntime.includes(props.runtime)) {
+      throw new Error('You can enable active tracing only for canaries that use version syn-nodejs-2.0 or later for their canary runtime.');
+    }
+
+    let memoryInMb = undefined;
+    if (!cdk.Token.isUnresolved(props.memory) && props.memory) {
+      memoryInMb = props.memory.toMebibytes();
+      if (memoryInMb % 64 !== 0) {
+        throw new Error(`\`memory\` must be a multiple of 64 MiB, got ${memoryInMb} MiB.`);
+      }
+      if (memoryInMb < 960 || memoryInMb > 3008) {
+        throw new Error(`\`memory\` must be between 960 MiB and 3008 MiB, got ${memoryInMb} MiB.`);
+      }
+    }
+
+    let timeoutInSeconds = undefined;
+    if (!cdk.Token.isUnresolved(props.timeout) && props.timeout) {
+      timeoutInSeconds = props.timeout.toSeconds();
```

Comment:
```suggestion
      const timeoutError = '`timeout` must be between 3 seconds and 840 seconds';
      const timeoutInMillis = props.timeout.toMilliseconds();
      if (timeoutInMillis > 0 && timeoutInMillis < Duration.seconds(3).toMilliseconds()) {
        throw new Error(`${timeoutError}, got ${timeoutInMillis} milliseconds.`);
      }
      timeoutInSeconds = props.timeout.toSeconds();
```
Handle exceptions when the user specifies timeout between 0 and 1 sec.

Reply:
@lpizzinidev 
Thank you. However, I did not quite understand the need for this validation. 
Could you please explain it in more detail?

From my testing, millisecond values that cannot be converted to whole seconds (e.g. 100ms, 3100ms, etc.) resulted in an error with the following validation:

https://github.com/aws/aws-cdk/blob/4e9ef15a0cf8b1bc664b5d22b34af9eaaa17bec8/packages/aws-cdk-lib/core/lib/duration.ts#L326

Therefore, millisecond values between 0-1 second would also result in an error with the above validation. 
As such, I considered that an additional check is unnecessary. What do you think?

Reply:
The point of adding the validation is to handle the error on the construct and provide a meaningful error message to the user.

Reply:
Thank you. I added validation to throw an error if the value is not an integer representing seconds. 
How about this? 
With this change, it will also throw an error if a value between 0 and 1 second is set.